src/registries/npm-registry.js

```diff
@@ -12,7 +12,7 @@ import {addSuffix, removePrefix} from '../util/misc';
 import isRequestToRegistry from './is-request-to-registry.js';
 
 const defaults = require('defaults');
-const userHome = require('user-home');
+const userHome = require('os').homedir();
```

Comment:
Considering that constants.js has custom edge case handling to homedir, can you extract that part into an internal module in `src/util`?
I think that edge case handling may cause some issues